lib/tiny_client/remote_client.rb

```diff
@@ -16,6 +16,19 @@ def get(path, params, id, name)
       }.merge!(@config.headers), @config.connect_timeout, @config.verbose)
     end
 
+    #    GET /<path>/<id>/<name>?<params>
+    # @param [Hash] data
+    # @raise [ResponseError] if the server respond with an error status (i.e 404, 500..)
+    # @return [Response]
+    def body_data_get(path, params, id, name, data)
+      url = @config.url_for(path, id, name, params)
+      verify_json(data)
+      CurbRequestor.perform_body_data_get(url, {
+        'Accept' => 'application/json',
+        'Content-Type' => 'application/json'
+      }.merge!(@config.headers), data.to_json, @config.connect_timeout, @config.verbose)
```

Comment:
The core of this PR is here.

Reply:
Yes, large data should always be in the body not the URL 🤦